pkg/auth/auth.go

```diff
@@ -84,9 +85,9 @@ func Login(ctx context.Context, systemContext *types.SystemContext, opts *LoginO
 		}
 		logrus.Debugf("registry not specified, default to the first registry %q from registries.conf", server)
 	} else {
-		server = getRegistryName(args[0])
+		server = trimScheme(args[0])
```

Comment:
If we are going to make the behavior option-configured, I’d prefer to only allow schemes in the old input format, and reject that in the new  one.

Also, should this allow tags/digests? (Alternatively, maybe c/image should reject that. Or perhaps both.)


Let’s not make exactly the same design mistake again: Let’s restrict the allowed input now, so that we can detect users mistakes, or assign some different semantics to it in the future.

Reply:
Yes, this check is now part of `getCredentials`.